d2/src/main/java/com/linkedin/d2/balancer/properties/ClusterPropertiesJsonSerializer.java

```diff
@@ -77,6 +77,17 @@ public ClusterProperties fromBytes(byte[] bytes) throws PropertySerializationExc
     return (T) map.get(key);
   }
 
+  @SuppressWarnings("unchecked")
+  private static <T> T mapGetOrDefault(Map<String, Object> map, String key, T defaultValue)
```

Comment:
Can we extract these common methods to `com.linkedin.d2.balancer.properties.util.PropertyUtil.java`?